### PR TITLE
Add support for bash completion file production

### PR DESCRIPTION
**Pull request by srfraser**
Use cobra's own tools for generating the shell script.

    ./taskcluster completions
    source bash_completion.sh

Or:

    ./taskcluster completions mycompletion.sh
    source mycompletion.sh

or moving the file into `/etc/bash_completion.d` on Linux and resetting the shell.

### CODE REVIEW

**Comment by srfraser** [comment]
Example script at https://gist.github.com/srfraser/fb05cf2041fe28f004cd997d50c5d470

File: cmds/completions/completions.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+func init() {
+	// Add the task subtree to the root.
+	use := "completions <filename (default:" + defaultFilename + ")>"
+	completionsCommand := &cobra.Command{
+		Short: "Provides bash completion script.",
+		RunE:  genCompletion,
+		Use:   use,
+	}
```
Comment (jonasfj) [inline cmds/completions/completions.go]: So could I do:
`$ sudo taskcluster completions /etc/bash_completion.d/taskcluster.sh`

to install these completions globally?

Maybe we could put that in the `--help` message, or even make the default filename on linux... :)

Comment (srfraser) [reply cmds/completions/completions.go]: I didn't necessarily want to make the default output file one you needed special rights to write to, but adding the instructions to the help sounds like a good idea. Macs are... special

Comment (jonasfj) [reply cmds/completions/completions.go]: oh, yeah, good point..

**Pull request merged**